Content.Shared/_Goobstation/Clothing/Systems/SharedSealableClothingSystem.cs

```diff
@@ -0,0 +1,446 @@
+        SubscribeLocalEvent<SealableClothingControlComponent, ToggleClothingAttemptEvent>(OnToggleClothingAttempt);
+    }
+
+    #region Events
+
+    /// <summary>
+    /// Toggles components on part when suit complete sealing process
+    /// </summary>
+    /// <param name="part"></param>
+    /// <param name="args"></param>
+    private void OnPartSealingComplete(Entity<SealableClothingComponent> part, ref ClothingPartSealCompleteEvent args)
+    {
+        _componentTogglerSystem.ToggleComponent(part, args.IsSealed);
+    }
+
+    /// <summary>
+    ///     Toggles components on control when suit complete sealing process
+    /// </summary>
+    private void OnControlSealingComplete(Entity<SealableClothingControlComponent> control, ref ClothingControlSealCompleteEvent args)
+    {
+        _componentTogglerSystem.ToggleComponent(control, args.IsSealed);
+    }
+
+    /// <summary>
+    /// Add/Remove wearer on clothing equip/unequip
+    /// </summary>
+    private void OnControlEquip(Entity<SealableClothingControlComponent> control, ref ClothingGotEquippedEvent args)
+    {
+        control.Comp.WearerEntity = args.Wearer;
+        Dirty(control);
+    }
+
+    private void OnControlUnequip(Entity<SealableClothingControlComponent> control, ref ClothingGotUnequippedEvent args)
+    {
+        control.Comp.WearerEntity = null;
+        Dirty(control);
+    }
+
+    /// <summary>
+    /// Removes seal action on component remove
+    /// </summary>
+    private void OnControlRemove(Entity<SealableClothingControlComponent> control, ref ComponentRemove args)
+    {
+        var comp = control.Comp;
+
+        _actionsSystem.RemoveAction(comp.SealActionEntity);
+    }
+
+    /// <summary>
+    /// Ensures seal action to wearer when it equip the seal control
+    /// </summary>
+    private void OnControlGetItemActions(Entity<SealableClothingControlComponent> control, ref GetItemActionsEvent args)
+    {
+        var (uid, comp) = control;
+
+        if (comp.SealActionEntity == null || args.SlotFlags != comp.RequiredControlSlot)
+            return;
+
+        args.AddAction(comp.SealActionEntity.Value);
+    }
+
+    /// <summary>
+    /// Adds unsealing verbs to sealing control allowing other users to unseal/seal clothing via stripping
+    /// </summary>
+    private void OnEquipmentVerb(Entity<SealableClothingControlComponent> control, ref GetVerbsEvent<Verb> args)
+    {
+        var (uid, comp) = control;
+        var user = args.User;
+
+        if (!args.CanComplexInteract)
+            return;
+
+        // Since sealing control in wearer's container system just won't show verb on args.CanAccess
+        if (!_interactionSystem.InRangeUnobstructed(user, uid))
+            return;
+
+        if (comp.WearerEntity == null ||
+            comp.WearerEntity != user && _actionBlockerSystem.CanInteract(comp.WearerEntity.Value, null))
+            return;
```

Comment:
```suggestion
        if (!args.CanComplexInteract
            // Since sealing control in wearer's container system just won't show verb on args.CanAccess
            || !_interactionSystem.InRangeUnobstructed(user, uid)
            || comp.WearerEntity == null
            || comp.WearerEntity != user
            && _actionBlockerSystem.CanInteract(comp.WearerEntity.Value, null))
            return;
```